internal/profiling/server.go

```diff
@@ -0,0 +1,24 @@
+package profiling
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+)
+
+// NewProfilingServer creates a new HTTP server for profiling.
+func NewProfilingServer(port int) *http.Server {
+	mux := http.NewServeMux()
+	srv := http.Server{
+		Addr:    fmt.Sprintf(":%d", port),
```

Comment:
if you forward this entire address as argument instead of the port it allows people to use a different (e.g. non exposed) network for debugging in production environments without security risk.